pkg/svc/workflow.go

```diff
@@ -100,43 +106,83 @@ func (s *workflowsrvc) Assign(ctx context.Context, w *workflow.AssignPayload) (e
 		return err
 	}
 
-	err = s.codesetStore.CreateWebhook(ctx, codeset, wfListener.URL)
+	webhookID, err := s.codesetStore.CreateWebhook(ctx, codeset, wfListener.URL)
 	if err != nil {
 		s.logger.Print(err)
 		return err
 	}
 
-	s.store.AddCodesetAssignment(ctx, w.Name, codeset)
+	s.store.AddCodesetAssignment(ctx, w.Name, &domain.AssignedCodeset{Codeset: codeset, WebhookID: webhookID})
 
-	err = s.backend.CreateWorkflowRun(ctx, w.Name, codeset)
+	err = s.backend.CreateWorkflowRun(ctx, s.logger, w.Name, codeset)
 	if err != nil {
 		s.logger.Print(err)
 		return err
 	}
 	return nil
 }
 
-// ListAssignments lists workflow assignments
+// Unassign a Workflow from a Codeset.
+func (s *workflowsrvc) Unassign(ctx context.Context, u *workflow.UnassignPayload) (err error) {
+	s.logger.Print("workflow.unassign")
+	codeset, err := s.codesetStore.Find(ctx, u.CodesetProject, u.CodesetName)
+	if err != nil {
+		s.logger.Print(err)
+		return workflow.MakeNotFound(err)
+	}
+	return s.unassignCodesetFromWorkflow(ctx, u.Name, codeset)
+}
+
+// ListAssignments lists Workflow assignments.
 func (s *workflowsrvc) ListAssignments(ctx context.Context, w *workflow.ListAssignmentsPayload) (res []*workflow.WorkflowAssignment, err error) {
+	s.logger.Print("workflow.listAssignments")
 	listeners := map[string]*domain.WorkflowListener{}
-	for wf, codesets := range s.store.GetAssignments(ctx, w.Name) {
+	for wf, acs := range s.store.GetAssignments(ctx, w.Name) {
```

Comment:
`acs`, really? Didn't you want to call it `ass`? ;-)

Reply:
`acs` is an acronym for assigned codesets

Reply:
It was a joke...

Reply:
`ass` would've looked better :D not to mention the occasional laughs it might have caused